File: common/src/main/java/dev/mayaqq/estrogen/registry/EstrogenSounds.java
package dev.mayaqq.estrogen.registry;


import com.teamresourceful.resourcefullib.common.registry.ResourcefulRegistries;
import com.teamresourceful.resourcefullib.common.registry.ResourcefulRegistry;
import dev.mayaqq.estrogen.Estrogen;
import net.minecraft.core.registries.BuiltInRegistries;
import net.minecraft.core.registries.Registries;
import net.minecraft.sounds.SoundEvent;
import uwu.serenity.critter.api.entry.RegistryEntry;
import uwu.serenity.critter.stdlib.Registrar;

@SuppressWarnings("unused")
public class EstrogenSounds {
    public static final Registrar<SoundEvent> SOUNDS = Registrar.create(Estrogen.MOD_ID, Registries.SOUND_EVENT);

    // Dash
    public static final RegistryEntry<SoundEvent> DASH =  SOUNDS.entry("dash", () -> SoundEvent.createVariableRangeEvent(Estrogen.id("dash"))).register();

    // Music Discs
    public static final RegistryEntry<SoundEvent> G03C = SOUNDS.entry("g03c", () -> SoundEvent.createVariableRangeEvent(Estrogen.id("g03c"))).register();

    // Ambient Music
    public static final RegistryEntry<SoundEvent> TRUST_YOURSELF = SOUNDS.entry("trust_yourself", () -> SoundEvent.createVariableRangeEvent(Estrogen.id("trust_yourself"))).register();
    public static final RegistryEntry<SoundEvent> AMPHITRITE = SOUNDS.entry("amphitrite", () -> SoundEvent.createVariableRangeEvent(Estrogen.id("amphitrite"))).register();
    public static final RegistryEntry<SoundEvent> AURUM_BERRY = SOUNDS.entry("aurum_berry", () -> SoundEvent.createVariableRangeEvent(Estrogen.id("aurum_berry"))).register();
    public static final RegistryEntry<SoundEvent> ESTROGEN_AMBIENT = SOUNDS.entry("estrogen_ambient", () -> SoundEvent.createVariableRangeEvent(Estrogen.id("estrogen_ambient"))).register();
    public static final RegistryEntry<SoundEvent> INFERRED_DREAMS = SOUNDS.entry("inferred_dreams", () -> SoundEvent.createVariableRangeEvent(Estrogen.id("inferred_dreams"))).register();
    public static final RegistryEntry<SoundEvent> SLEEPING = SOUNDS.entry("sleeping", () -> SoundEvent.createVariableRangeEvent(Estrogen.id("sleeping"))).register();
    public static final RegistryEntry<SoundEvent> INNER_SELF_REALIZATION = SOUNDS.entry("inner_selfrealization", () -> SoundEvent.createVariableRangeEvent(Estrogen.id("inner_selfrealization"))).register();

    // Dream Blocks
    public static final RegistryEntry<SoundEvent> DREAM_BLOCK_ENTER = SOUNDS.entry("dream_block_enter", () -> SoundEvent.createVariableRangeEvent(Estrogen.id("dream_block_enter"))).register();
    public static final RegistryEntry<SoundEvent> DREAM_BLOCK_LOOP = SOUNDS.entry("dream_block_loop", () -> SoundEvent.createVariableRangeEvent(Estrogen.id("dream_block_loop"))).register();
    public static final RegistryEntry<SoundEvent> DREAM_BLOCK_EXIT = SOUNDS.entry("dream_block_exit", () -> SoundEvent.createVariableRangeEvent(Estrogen.id("dream_block_exit"))).register();
    public static final RegistryEntry<SoundEvent> DREAM_BLOCK_PLACE = SOUNDS.entry("dream_block_place", () -> SoundEvent.createVariableRangeEvent(Estrogen.id("dream_block_place"))).register();
    public static final RegistryEntry<SoundEvent> DREAM_BLOCK_BREAK = SOUNDS.entry("dream_block_break", () -> SoundEvent.createVariableRangeEvent(Estrogen.id("dream_block_break"))).register();
    public static final RegistryEntry<SoundEvent> DREAM_BLOCK_STEP = SOUNDS.entry("dream_block_step", () -> SoundEvent.createVariableRangeEvent(Estrogen.id("dream_block_step"))).register();
    public static final RegistryEntry<SoundEvent> DREAM_BLOCK_FALL = SOUNDS.entry("dream_block_fall", () -> SoundEvent.createVariableRangeEvent(Estrogen.id("dream_block_fall"))).register();
    public static final RegistryEntry<SoundEvent> DREAM_BLOCK_HIT = SOUNDS.entry("dream_block_hit", () -> SoundEvent.createVariableRangeEvent(Estrogen.id("dream_block_hit"))).register();

    // Dormant Dream Block
    public static final RegistryEntry<SoundEvent> DREAM_BLOCK_DORMANT_BREAK = SOUNDS.entry("dream_block_dormant_break", () -> SoundEvent.createVariableRangeEvent(Estrogen.id("dream_block_dormant_break"))).register();
    public static final RegistryEntry<SoundEvent> DREAM_BLOCK_DORMANT_PLACE = SOUNDS.entry("dream_block_dormant_place", () -> SoundEvent.createVariableRangeEvent(Estrogen.id("dream_block_dormant_place"))).register();
    public static final RegistryEntry<SoundEvent> DREAM_BLOCK_DORMANT_STEP = SOUNDS.entry("dream_block_dormant_step", () -> SoundEvent.createVariableRangeEvent(Estrogen.id("dream_block_dormant_step"))).register();
    public static final RegistryEntry<SoundEvent> DREAM_BLOCK_DORMANT_FALL = SOUNDS.entry("dream_block_dormant_fall", () -> SoundEvent.createVariableRangeEvent(Estrogen.id("dream_block_dormant_fall"))).register();
    public static final RegistryEntry<SoundEvent> DREAM_BLOCK_DORMANT_HIT = SOUNDS.entry("dream_block_dormant_hit", () -> SoundEvent.createVariableRangeEvent(Estrogen.id("dream_block_dormant_hit"))).register();

    // Cookie Jar
    public static final RegistryEntry<SoundEvent> JAR_FULL = SOUNDS.entry("jar_full", () -> SoundEvent.createVariableRangeEvent(Estrogen.id("jar_full"))).register();
    public static final RegistryEntry<SoundEvent> JAR_PLACE = SOUNDS.entry("jar_place", () -> SoundEvent.createVariableRangeEvent(Estrogen.id("jar_place"))).register();
    public static final RegistryEntry<SoundEvent> JAR_BREAK = SOUNDS.entry("jar_break", () -> SoundEvent.createVariableRangeEvent(Estrogen.id("jar_break"))).register();
    public static final RegistryEntry<SoundEvent> JAR_STEP = SOUNDS.entry("jar_step", () -> SoundEvent.createVariableRangeEvent(Estrogen.id("jar_step"))).register();
    public static final RegistryEntry<SoundEvent> JAR_INSERT = SOUNDS.entry("jar_insert", () -> SoundEvent.createVariableRangeEvent(Estrogen.id("jar_insert"))).register();
    public static final RegistryEntry<SoundEvent> JAR_HIT = SOUNDS.entry("jar_hit", () -> SoundEvent.createVariableRangeEvent(Estrogen.id("jar_hit"))).register();
    public static final RegistryEntry<SoundEvent> JAR_FALL = SOUNDS.entry("jar_fall", () -> SoundEvent.createVariableRangeEvent(Estrogen.id("jar_fall"))).register();

    // Pill Box
    public static final RegistryEntry<SoundEvent> PILL_BOX_PLACE = SOUNDS.entry("pill_box_place", () -> SoundEvent.createVariableRangeEvent(Estrogen.id("pill_box_place"))).register();
    public static final RegistryEntry<SoundEvent> PILL_BOX_BREAK = SOUNDS.entry("pill_box_break", () -> SoundEvent.createVariableRangeEvent(Estrogen.id("pill_box_break"))).register();
    public static final RegistryEntry<SoundEvent> PILL_BOX_STEP = SOUNDS.entry("pill_box_step", () -> SoundEvent.createVariableRangeEvent(Estrogen.id("pill_box_step"))).register();
    public static final RegistryEntry<SoundEvent> PILL_BOX_HIT = SOUNDS.entry("pill_box_hit", () -> SoundEvent.createVariableRangeEvent(Estrogen.id("pill_box_hit"))).register();
    public static final RegistryEntry<SoundEvent> PILL_BOX_FALL = SOUNDS.entry("pill_box_fall", () -> SoundEvent.createVariableRangeEvent(Estrogen.id("pill_box_fall"))).register();

    // Moth
    public static final RegistryEntry<SoundEvent> MOTH_DEATH = SOUNDS.entry("moth_death", () -> SoundEvent.createVariableRangeEvent(Estrogen.id("moth_death"))).register();
    public static final RegistryEntry<SoundEvent> MOTH_HURT = SOUNDS.entry("moth_hurt", () -> SoundEvent.createVariableRangeEvent(Estrogen.id("moth_hurt"))).register();
    public static final RegistryEntry<SoundEvent> MOTH_LOOP = SOUNDS.entry("moth_loop", () -> SoundEvent.createVariableRangeEvent(Estrogen.id("moth_loop"))).register();
    public static final RegistryEntry<SoundEvent> MOTH_FUZZ_UP = SOUNDS.entry("moth_fuzz_up", () -> SoundEvent.createVariableRangeEvent(Estrogen.id("moth_fuzz_up"))).register();
}

File: src/main/kotlin/dev/mayaqq/estrogen/content/EstrogenSounds.kt
@file:Suppress("unused")

package dev.mayaqq.estrogen.content

import dev.mayaqq.estrogen.MOD_ID
import invoke.kitty.kritter.registry.api.Registrar
import invoke.kitty.kritter.registry.api.entry.RegistryEntry
import invoke.kitty.kritter.registry.misc.soundEvent
import invoke.kitty.kritter.utils.ExperimentalRegistryApi
import invoke.kitty.kritter.utils.sounds.LazySoundType
import net.minecraft.core.registries.Registries
import net.minecraft.sounds.SoundEvent
import net.minecraft.world.level.block.SoundType

@OptIn(ExperimentalRegistryApi::class)
object EstrogenSounds : Registrar<SoundEvent> by Registrar(MOD_ID, Registries.SOUND_EVENT) {

    // Dash
    val DASH: RegistryEntry<SoundEvent> = soundEvent("dash")

    // Music disc
    val G03C: RegistryEntry<SoundEvent> = soundEvent("g03c")

    // Ambient music
    val TRUST_YOURSELF: RegistryEntry<SoundEvent> = soundEvent("trust_yourself")
    val AMPHITRITE: RegistryEntry<SoundEvent> = soundEvent("amphitrite")
    val AURUM_BERRY: RegistryEntry<SoundEvent> = soundEvent("aurum_berry")
    val ESTROGEN_AMBIENT: RegistryEntry<SoundEvent> = soundEvent("estrogen_ambient")
    val INFERRED_DREAMS: RegistryEntry<SoundEvent> = soundEvent("inferred_dream")
    val SLEEPING: RegistryEntry<SoundEvent> = soundEvent("sleeping")
    val INNER_SELF_REALIZATION: RegistryEntry<SoundEvent> = soundEvent("inner_self_realization")

    // Dream block
    val DREAM_BLOCK_ENTER: RegistryEntry<SoundEvent> = soundEvent("dream_block_enter")
    val DREAM_BLOCK_EXIT: RegistryEntry<SoundEvent> = soundEvent("dream_block_exit")
    val DREAM_BLOCK_LOOP: RegistryEntry<SoundEvent> = soundEvent("dream_block_loop")
    val DREAM_BLOCK_PLACE: RegistryEntry<SoundEvent> = soundEvent("dream_block_place")
    val DREAM_BLOCK_BREAK: RegistryEntry<SoundEvent> = soundEvent("dream_block_break")
    val DREAM_BLOCK_STEP: RegistryEntry<SoundEvent> = soundEvent("dream_block_step")
    val DREAM_BLOCK_HIT: RegistryEntry<SoundEvent> = soundEvent("dream_block_hit")
    val DREAM_BLOCK_FALL: RegistryEntry<SoundEvent> = soundEvent("dream_block_fall")

    // Dormant dream block
    val DREAM_BLOCK_DORMANT_PLACE: RegistryEntry<SoundEvent> = soundEvent("dream_block_dormant_place")
    val DREAM_BLOCK_DORMANT_BREAK: RegistryEntry<SoundEvent> = soundEvent("dream_block_dormant_break")
    val DREAM_BLOCK_DORMANT_STEP: RegistryEntry<SoundEvent> = soundEvent("dream_block_dormant_step")
    val DREAM_BLOCK_DORMANT_HIT: RegistryEntry<SoundEvent> = soundEvent("dream_block_dormant_hit")
    val DREAM_BLOCK_DORMANT_FALL: RegistryEntry<SoundEvent> = soundEvent("dream_block_dormant_fall")

    // Cookie jar
    val JAR_FULL: RegistryEntry<SoundEvent> = soundEvent("jar_full")
    val JAR_INSERT: RegistryEntry<SoundEvent> = soundEvent("jar_insert")
    val JAR_PLACE: RegistryEntry<SoundEvent> = soundEvent("jar_place")
    val JAR_BREAK: RegistryEntry<SoundEvent> = soundEvent("jar_break")
    val JAR_STEP: RegistryEntry<SoundEvent> = soundEvent("jar_step")
    val JAR_HIT: RegistryEntry<SoundEvent> = soundEvent("jar_hit")
    val JAR_FALL: RegistryEntry<SoundEvent> = soundEvent("jar_fall")

    // Estrogen Pill Box
    val PILL_BOX_PLACE: RegistryEntry<SoundEvent> = soundEvent("pill_box_place")
    val PILL_BOX_STEP: RegistryEntry<SoundEvent> = soundEvent("pill_box_step")
    val PILL_BOX_BREAK: RegistryEntry<SoundEvent> = soundEvent("pill_box_break")
    val PILL_BOX_HIT: RegistryEntry<SoundEvent> = soundEvent("pill_box_hit")
    val PILL_BOX_FALL: RegistryEntry<SoundEvent> = soundEvent("pill_box_fall")

    val MOTH_DEATH: RegistryEntry<SoundEvent> = soundEvent("moth_death")
    val MOTH_HURT: RegistryEntry<SoundEvent> = soundEvent("moth_hurt")
    val MOTH_LOOP: RegistryEntry<SoundEvent> = soundEvent("moth_loop")
    val MOTH_FUZZ_UP: RegistryEntry<SoundEvent> = soundEvent("moth_fuzz_up")
}

object EstrogenSoundTypes {

    val DREAM_BLOCK_ACTIVE: SoundType = LazySoundType(1.0f, 1.0f,
        breakSound = EstrogenSounds.DREAM_BLOCK_BREAK,
        stepSound = EstrogenSounds.DREAM_BLOCK_STEP,
        placeSound = EstrogenSounds.DREAM_BLOCK_PLACE,
        hitSound = EstrogenSounds.DREAM_BLOCK_HIT,
        fallSound = EstrogenSounds.DREAM_BLOCK_FALL
    )

    val DREAM_BLOCK_DORMANT: SoundType = LazySoundType(1.0f, 1.0f,
        breakSound = EstrogenSounds.DREAM_BLOCK_BREAK,
        stepSound = EstrogenSounds.DREAM_BLOCK_STEP,
        placeSound = EstrogenSounds.DREAM_BLOCK_PLACE,
        hitSound = EstrogenSounds.DREAM_BLOCK_DORMANT_HIT,
        fallSound = EstrogenSounds.DREAM_BLOCK_DORMANT_FALL
    )

    val COOKIE_JAR: SoundType = LazySoundType(1.0f, 1.0f,
        breakSound = EstrogenSounds.JAR_BREAK,
        stepSound = EstrogenSounds.JAR_STEP,
        placeSound = EstrogenSounds.JAR_PLACE,
        hitSound = EstrogenSounds.JAR_HIT,
        fallSound = EstrogenSounds.JAR_FALL
    )

    val PILL_BOX: SoundType = LazySoundType(1.0f, 1.0f,
        breakSound = EstrogenSounds.PILL_BOX_BREAK,
        stepSound = EstrogenSounds.PILL_BOX_STEP,
        placeSound = EstrogenSounds.PILL_BOX_PLACE,
        hitSound = EstrogenSounds.PILL_BOX_HIT,
        fallSound = EstrogenSounds.PILL_BOX_FALL
    )
}